src/panel/context/Request.tsx

```diff
@@ -26,41 +26,45 @@ export const RequestContext = createContext<RequestContextValue>(null as any);
 
 export const RequestProvider: FC = ({ children }) => {
   const { sendMessage, addMessageHandler } = useContext(DevtoolsContext);
-  const [fetching, setFetching] = useState(false);
+  const [state, setState] = useState<{
+    fetching: boolean;
+    response?: object;
+    error?: object;
+  }>({ fetching: false, response: undefined, error: undefined });
   const [query, setQuery] = useState<string>();
-  const [response, setResponse] = useState<object | undefined>();
-  const [error, setError] = useState<object | undefined>();
   const [schema, setSchema] = useState<GraphQLSchema>();
 
   const execute = useCallback(() => {
-    setFetching(true);
-    setResponse(undefined);
-    setError(undefined);
+    setState({
+      fetching: true,
+      response: undefined,
+      error: undefined
+    });
     sendMessage({ type: "request", query: query || "" });
   }, [query, sendMessage]);
 
   // Listen for response for devtools
   useEffect(() => {
     return addMessageHandler(e => {
-      if (
-        !fetching ||
-        e.type === "operation" ||
-        e.type === "init" ||
-        e.type === "disconnect" ||
-        (e.data.operation.context.meta as any).source !== "Devtools"
-      ) {
-        return;
-      }
-
-      if (e.data.error !== undefined) {
-        setError(e.data.error);
-      } else {
-        setResponse(e.data.data);
-      }
+      setState(s => {
+        if (
+          !s.fetching ||
```

Comment:
So we ignore everything but the connection message here, can't we simplify this check by inversing it (this assumes that the connection message has some descriptive property for this)

Reply:
I think I did this for something to do with TS types - fixed :+1: